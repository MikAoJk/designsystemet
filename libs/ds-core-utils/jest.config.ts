/* eslint-disable */
export default {
  displayName: 'ds-core-utils',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-core-utils',
};
