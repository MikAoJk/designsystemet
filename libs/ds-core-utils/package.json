{
  "name": "@skatteetaten/ds-core-utils",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.0.1",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": ">=17"
  }
}
