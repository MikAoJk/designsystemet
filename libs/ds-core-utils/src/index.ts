export * from './SymbolLogo/SymbolLogo';
export * from './base-props.types';
export * from './defaults';
export * from './example-layouts/ExternalLayout';
export * from './example-layouts/InternalLayout';
export * from './i18n';
export * from './Separator/Separator';
