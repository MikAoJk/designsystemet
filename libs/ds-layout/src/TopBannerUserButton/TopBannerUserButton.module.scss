@import 'libs/ds-core-designtokens/lib/designtokens/breakpoints';

.username {
  &Button {
    max-width: 5.25rem;

    @media (min-width: $breakpoint-xs) {
      max-width: 6.25rem;
    }

    @media (min-width: $breakpoint-s) {
      max-width: 7.5rem;
    }

    @media (min-width: $breakpoint-m) {
      max-width: 20rem;
    }
  }

  &Text {
    overflow: hidden;
    width: 100%;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
}
