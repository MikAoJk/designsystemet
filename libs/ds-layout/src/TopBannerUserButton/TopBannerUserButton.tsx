import { forwardRef, JSX } from 'react';
import { useTranslation } from 'react-i18next';

import { dsI18n, getCommonClassNameDefault } from '@skatteetaten/ds-core-utils';
import {
  AccountMultipleSVGpath,
  BriefcaseSVGpath,
  PersonSVGpath,
} from '@skatteetaten/ds-icons';

import { TopBannerUserButtonProps } from './TopBannerUserButton.types';
import { TopBannerButton } from '../TopBannerButton/TopBannerButton';

import styles from './TopBannerUserButton.module.scss';

export const TopBannerUserButton = forwardRef<
  HTMLButtonElement,
  TopBannerUserButtonProps
>(
  (
    {
      id,
      className = getCommonClassNameDefault(),
      lang,
      'data-testid': dataTestId,
      role,
      children,
      onUsername,
    },
    ref
  ): JSX.Element => {
    const { t } = useTranslation('ds_pages', { i18n: dsI18n });

    let buttonText = children;
    let roleSVGpath = PersonSVGpath;
    let title = '';
    if (role === 'meg') {
      buttonText = t('topbannerbutton.Myself');
      roleSVGpath = PersonSVGpath;
      title = t('topbannerbutton.MyselfTitle');
    } else if (role === 'virksomhet') {
      roleSVGpath = BriefcaseSVGpath;
      title = t('topbannerbutton.CompanyTitle');
    } else if (role === 'andre') {
      roleSVGpath = AccountMultipleSVGpath;
      title = t('topbannerbutton.OtherTitle');
    }

    return (
      <TopBannerButton
        ref={ref}
        id={id}
        className={className}
        classNames={
          role === 'meg'
            ? undefined
            : {
                container: styles.usernameButton,
                text: styles.usernameText,
              }
        }
        lang={lang}
        data-testid={dataTestId}
        variant={'filled'}
        svgPath={roleSVGpath}
        title={title}
        onClick={onUsername}
      >
        {buttonText}
      </TopBannerButton>
    );
  }
);

TopBannerUserButton.displayName = 'TopBannerUserButton';
