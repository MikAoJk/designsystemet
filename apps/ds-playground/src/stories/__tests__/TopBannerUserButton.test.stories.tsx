import { dsI18n } from '@skatteetaten/ds-core-utils';
import { expect } from '@storybook/jest';
import { Meta, StoryFn, StoryObj } from '@storybook/react';
import { within } from '@storybook/testing-library';

// eslint-disable-next-line @nx/enforce-module-boundaries
import { TopBannerUserButton } from '../../../../../libs/ds-layout/src/TopBannerUserButton/TopBannerUserButton';
// eslint-disable-next-line @nx/enforce-module-boundaries
import {
  TopBannerUserButtonProps,
  userRoleArr,
} from '../../../../../libs/ds-layout/src/TopBannerUserButton/TopBannerUserButton.types';

const verifyAttribute =
  (attribute: string, expectedValue: string) =>
  async ({ canvasElement }: { canvasElement: HTMLElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole('button');
    await expect(button).toBeInTheDocument();
    await expect(button).toHaveAttribute(attribute, expectedValue);
  };

const meta = {
  component: TopBannerUserButton,
  title: 'Tester/TopBannerUserButton (intern)',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    role: {
      table: { disable: true },
      options: [...userRoleArr],
      control: 'inline-radio',
    },
    // Events
    onClick: { table: { disable: true } },
  },
} as Meta<typeof TopBannerUserButton>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs: TopBannerUserButtonProps = {
  role: 'meg',
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLButtonElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: verifyAttribute('id', 'dummyIdForwardedFromRef'),
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole('button');
    await expect(button).toHaveAttribute('id', 'htmlid');
    await expect(button).toHaveClass('dummyClassname');
    await expect(button).toHaveAttribute('lang', 'nb');
    await expect(button).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (Username A2)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    role: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    // TODO FRONT 1161 - erstattet hardkodet tekst når tekstvariabelen finnes
    const usernameText = canvas.getByText('Meg selv');
    const defaultTitle = canvas.getByText(
      dsI18n.t('ds_pages:userbutton.SelfIconText')
    );
    await expect(usernameText).toBeInTheDocument();
    await expect(defaultTitle).toBeInTheDocument();
  },
} satisfies Story;

// TODO FRONT 1161 skrive test WithChildren

const TemplateWithAllRoles: StoryFn<typeof TopBannerUserButton> = (args) => (
  <div className={'topbannerButtonContainer'}>
    {userRoleArr.map((role, index) => {
      return (
        <TopBannerUserButton
          key={`user_${index}`}
          className={'bottomSpacingXL'}
          {...args}
          role={role}
        >
          {`${
            role === 'andre'
              ? 'SØRUMSAND-SCHOENBERG WILLIAM'
              : 'Navn på virksomhet'
          }`}
        </TopBannerUserButton>
      );
    })}
  </div>
);

export const WithAllRoles = {
  render: TemplateWithAllRoles,
  name: 'With All Role (Username A2, A6, A7)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    role: {
      table: { disable: false },
      control: { type: null },
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const titleSelf = canvas.getByText(
      dsI18n.t('ds_pages:userbutton.SelfIconText')
    );
    const titleWork = canvas.getByText(
      dsI18n.t('ds_pages:userbutton.WorkIconText')
    );
    const titleOther = canvas.getByText(
      dsI18n.t('ds_pages:userbutton.OtherIconText')
    );
    await expect(titleSelf).toBeInTheDocument();
    await expect(titleWork).toBeInTheDocument();
    await expect(titleOther).toBeInTheDocument();
  },
} satisfies Story;

export const WithMobileScreen = {
  render: TemplateWithAllRoles,
  name: 'With Small Screen (Username A2)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--mobile',
    },
  },
} satisfies Story;

export const WithBreakpointXS = {
  render: TemplateWithAllRoles,
  name: 'With Breakpoint-xs (Username A2)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xs',
    },
  },
} satisfies Story;

export const WithBreakpointS = {
  render: TemplateWithAllRoles,
  name: 'With Breakpoint-s (Username A2)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-s',
    },
  },
} satisfies Story;

export const WithBreakpointM = {
  render: TemplateWithAllRoles,
  name: 'With Breakpoint-m (Username A2)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-m',
    },
  },
} satisfies Story;

export const WithBreakpointL = {
  render: TemplateWithAllRoles,
  name: 'With Breakpoint-l (Username A2)',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-l',
    },
  },
} satisfies Story;
