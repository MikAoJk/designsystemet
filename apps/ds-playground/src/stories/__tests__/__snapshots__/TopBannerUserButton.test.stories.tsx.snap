// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/TopBannerUserButton (intern) Defaults play-test 1`] = `
<div data-test-block="true">
  <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx"
          type="button"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r2:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r2:">
        Innlogget som
      </title>
      <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
      </path>
    </svg>
    <span class>
      Meg selv
    </span>
  </button>
</div>
`;

exports[`Tester/TopBannerUserButton (intern) WithAllRoles play-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r4:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r4:">
          Innlogget som
        </title>
        <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
        </path>
      </svg>
      <span class>
        Meg selv
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r5:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r5:">
          Innlogget som annen
        </title>
        <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r6:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r6:">
          Innlogget som virksomhet
        </title>
        <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBannerUserButton (intern) WithAttributes play-test 1`] = `
<div data-test-block="true">
  <button id="htmlid"
          class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx dummyClassname"
          lang="nb"
          data-testid="123ID"
          type="button"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r1:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r1:">
        Innlogget som
      </title>
      <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
      </path>
    </svg>
    <span class>
      Meg selv
    </span>
  </button>
</div>
`;

exports[`Tester/TopBannerUserButton (intern) WithBreakpointL smoke-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rj:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rj:">
          Innlogget som
        </title>
        <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
        </path>
      </svg>
      <span class>
        Meg selv
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rk:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rk:">
          Innlogget som annen
        </title>
        <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rl:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rl:">
          Innlogget som virksomhet
        </title>
        <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBannerUserButton (intern) WithBreakpointM smoke-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rg:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rg:">
          Innlogget som
        </title>
        <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
        </path>
      </svg>
      <span class>
        Meg selv
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rh:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rh:">
          Innlogget som annen
        </title>
        <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:ri:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:ri:">
          Innlogget som virksomhet
        </title>
        <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBannerUserButton (intern) WithBreakpointS smoke-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rd:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rd:">
          Innlogget som
        </title>
        <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
        </path>
      </svg>
      <span class>
        Meg selv
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:re:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:re:">
          Innlogget som annen
        </title>
        <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rf:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rf:">
          Innlogget som virksomhet
        </title>
        <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBannerUserButton (intern) WithBreakpointXS smoke-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:ra:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:ra:">
          Innlogget som
        </title>
        <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
        </path>
      </svg>
      <span class>
        Meg selv
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rb:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rb:">
          Innlogget som annen
        </title>
        <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rc:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rc:">
          Innlogget som virksomhet
        </title>
        <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBannerUserButton (intern) WithChildren play-test 1`] = `
<div data-test-block="true">
  <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx  TopBannerUserButton_usernameButton__06VaY"
          type="button"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r3:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r3:">
        Innlogget som virksomhet
      </title>
      <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
      </path>
    </svg>
    <span class="TopBannerUserButton_usernameText__u85YY">
      Navnet på virksomheten
    </span>
  </button>
</div>
`;

exports[`Tester/TopBannerUserButton (intern) WithMobileScreen smoke-test 1`] = `
<div data-test-block="true">
  <div class="topbannerButtonContainer">
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r7:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r7:">
          Innlogget som
        </title>
        <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
        </path>
      </svg>
      <span class>
        Meg selv
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r8:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r8:">
          Innlogget som annen
        </title>
        <path d="M15.94 14c5.735.35 6.043 3.62 6.06 3.97V20h-4v-2a5.321 5.321 0 0 0-2.06-4ZM9 14c6.65 0 6.982 3.61 7 3.971V20H2v-2.029C2.019 17.61 2.35 14 9 14Zm6-9a3.5 3.5 0 1 1 0 7 3.39 3.39 0 0 1-1.93-.59 5.001 5.001 0 0 0 0-5.82A3.39 3.39 0 0 1 15 5ZM9 5a3.5 3.5 0 1 1 0 7 3.5 3.5 0 0 1 0-7Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        SØRUMSAND-SCHOENBERG WILLIAM
      </span>
    </button>
    <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx bottomSpacingXL TopBannerUserButton_usernameButton__06VaY"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:r9:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:r9:">
          Innlogget som virksomhet
        </title>
        <path d="M10 2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8c0-1.11.89-2 2-2h4V4c0-1.11.89-2 2-2Zm4 4V4h-4v2h4Z">
        </path>
      </svg>
      <span class="TopBannerUserButton_usernameText__u85YY">
        Navn på virksomhet
      </span>
    </button>
  </div>
</div>
`;

exports[`Tester/TopBannerUserButton (intern) WithRef play-test 1`] = `
<div data-test-block="true">
  <button class="TopBannerButton_button__5RUa7 TopBannerButton_button_filled__oJNLx"
          type="button"
          id="dummyIdForwardedFromRef"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz TopBannerButton_icon__FyVd5"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r0:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r0:">
        Innlogget som
      </title>
      <path d="M12 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z">
      </path>
    </svg>
    <span class>
      Meg selv
    </span>
  </button>
</div>
`;
